extensions/warp-ipfs/src/store/message/task.rs

```diff
@@ -2157,12 +2166,19 @@ impl ConversationTask {
             ConversationImageType::Icon => MAX_CONVERSATION_ICON_SIZE,
         };
 
-        let own_did = self.identity.did_key();
+        let Some(creator) = self.document.creator.clone() else {
+            return Err(Error::InvalidConversation);
+        };
+
+        let own_did = &self.identity.did_key();
 
-        if self.document.conversation_type() == ConversationType::Group
-            && !matches!(self.document.creator.as_ref(), Some(creator) if own_did.eq(creator))
+        if !&self
+            .document
+            .permissions
+            .has_permission(own_did, GroupPermission::EditGroupInfo)
+            && creator.ne(own_did)
```

Comment:
We need to have a check here to determine if the conversation is direct or group.

Reply:
ok. wasnt sure cause previously if it were a direct one it just let it pass through